Redirect Studio app users to magic login flow when signing up

SiobhyB opened the pull request:
Related to 5531-gh-Automattic/dotcom-forge, 5532-gh-Automattic/dotcom-forge 

## Proposed Changes

* With this PR, users who sign up for WordPress.com via the Studio app will be redirected to the magic login form. 
* This change will prevent a poor experience, in which users who signed up following the Studio flow were not correctly logged into their WordPress.com account. It will also avoid users being directed through the site creation flow. Further context for this decision can be found [here](5531-gh-Automattic/dotcom-forge#issuecomment-2084811812).

| Signup form  | Confirmation page |
| ------------- | ------------- |
| <img src="https://github.com/Automattic/wp-calypso/assets/2998162/49ed91f9-7f74-4499-987e-140e4ec23ae5" width="100%"> | <img src="https://github.com/Automattic/wp-calypso/assets/2998162/11e5bda2-783e-400b-9ad4-aae1e0ee73e2" width="100%"> |

>  [!NOTE]
> There is a `Create free account` link in the Studio app that will be updated after this PR has been merged.

## Testing Instructions

* Log out of WordPress.com and navigate to [the Calypso Live link in this comment](https://github.com/Automattic/wp-calypso/pull/90204#issuecomment-2090154172).
* Add the following to the end of the Calypso Live link in order to simular the URL that's clicked from the Studio app: `/log-in?client_id=95109&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fresponse_type%3Dtoken%26client_id%3D95109%26redirect_uri%3Dwpcom-local-dev%253A%252F%252Fauth%26scope%3Dglobal%26from-calypso%3D1`
* Select the option to create a new account from the login page.
* Submit an email address that's not already associated with a WordPress.com account.
* Click the verification link in the new email and verify that you're taken to authentication page for the Studio app.
* Confirm the following tracks events have been logged:
   * `calypso_login_block_login_form_send_account_create_magic_link` with a `client_id: 95109` property
   * `calypso_login_email_link_success` with a `client_id: 95109` property 

## Pre-merge Checklist

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [x] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [x] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [x] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [x] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [x] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

SiobhyB committed:
fix: Redirect Studio signups to magic login page

SiobhyB committed:
feat: Enhance magic login page for Studio signups

SiobhyB commented:
@matt-west, 👋 , this PR's not quite ready for review yet, but I wanted to get your early thoughts from a design perspective. The default styles being used are the same as those found on the magic login page, as you can see here:

<img width="541" alt="Screenshot 2024-05-02 at 22 44 49" src="https://github.com/Automattic/wp-calypso/assets/2998162/ae3cbe1f-44ed-4a2e-b7e8-aaa70ed76d90">

But, the styles and text can be changed based on what you think is best. You can test the proposed flow by:

* Logging out of WordPress.com and navigating directly to: `https://container-kind-ellis.calypso.live/log-in?client_id=95109&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fresponse_type%3Dtoken%26client_id%3D95109%26redirect_uri%3Dwpcom-local-dev%253A%252F%252Fauth%26scope%3Dglobal%26from-calypso%3D1` (simulating the URL clicked from the Studio app)
* Select the option to create a new account.
* Submit an email address that's not already associated with a WordPress.com account.
* Click the verification link in the new email.

Looking forward to hearing your initial impressions as well any iterations for the design and/or overall flow.

matt-west commented:
Thanks for the ping @SiobhyB!

I’m getting an error when I try to test that URL: `Container kind-ellis not found `
Perhaps it got cleaned up?

We should stick with the default WordPress.com styling for consistency.

SiobhyB commented:
@matt-west, ah, yes, I see a new live link was generated since my ping, this one should work: https://container-interesting-ptolemy.calypso.live/log-in?client_id=95109&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fresponse_type%3Dtoken%26client_id%3D95109%26redirect_uri%3Dwpcom-local-dev%253A%252F%252Fauth%26scope%3Dglobal%26from-calypso%3D1 

> We should stick with the default WordPress.com styling for consistency.

Perfect! If the current design works I'll continue as-is, but let me know if you think any tweaks are needed to the copy or anything else, too. :)

SiobhyB committed:
refactor: Update copy to be used on signup page

SiobhyB committed:
Merge branch 'trunk' into studio/magic-email-signup-flow

SiobhyB committed:
refactor: Follow Gravatar's pattern for Studio

SiobhyB commented:
@matt-west, thank you for the guidance! I've set this as ready for review now.

matt-west commented:
Looks good! 👍 Thanks @SiobhyB

<img width="671" alt="Screenshot 2024-05-07 at 16 54 09" src="https://github.com/Automattic/wp-calypso/assets/417538/e3faa264-5bb4-414d-bfe4-35200ba2fd32">

SiobhyB commented:
Temporarily setting this back to draft as I noticed an issue with the link, I'll change it back as ready for review when fixed...

SiobhyB committed:
Merge branch 'trunk' into studio/magic-email-signup-flow

SiobhyB committed:
Merge branch 'trunk' into studio/magic-email-signup-flow

dcalhoun commented on client/login/magic-login/style.scss:
```diff
@@ -288,6 +288,18 @@
+	text-align: left;
+
+	a {
+		@include breakpoint-deprecated( ">480px" ) {
```
Do we know why this mixin is deprecated? Is there an alternative we should use instead?

SiobhyB replied:
Thank you for pointing this out, David! I found in https://github.com/Automattic/wp-calypso/pull/41985 that this was replaced with `@include break-mobile`, which I see is used throughout this CSS file too. Updated in https://github.com/Automattic/wp-calypso/pull/90204/commits/483625912c31dd4b537bb1f9b5a2f8ad868d52d1.

dcalhoun commented on client/login/magic-login/index.jsx:
```diff
@@ -477,6 +505,30 @@ class MagicLogin extends Component {
 			);
 		}
 
+		// "query?.redirect_to" is used to determine if Studio app users are creating a new account (vs. logging in)
```
I do not follow this comment. From my testing, the `redirect_to` parameter is present for both logging in and account creation. 

What testing steps might I take to result in an outcome different that what is outlined in the PR description?

SiobhyB replied:
@dcalhoun, to make sure I understand: are you seeing the `redirect_to` option when you choose "Email me a login link" from the main login page? That shouldn't be the case, the `redirect_to` option should only be used when choosing the options to create an account (it does also appear on the _main_ login page, but that isn't being targeted with the magic login code): 

https://github.com/Automattic/wp-calypso/assets/2998162/f397bc1f-1060-4432-b0a1-1cdbc7e04254

dcalhoun replied:
Your screen recording and comment helped resolve my original confusion. I was seeking how I might arrive to a place where the `redirect_to` parameter was absent. As illustrated in my screenshot below, I now understand there are two URLs — `/log-in` and `/log-in/link` — and the `redirect_to` parameter comment on this line refers to its presence in the latter URL. 

<img width="1548" alt="log-in-url-parameters" src="https://github.com/Automattic/wp-calypso/assets/438664/c5b05872-76c6-4d17-8b91-0c88980570c7">


> are you seeing the `redirect_to` option when you choose "Email me a login link" from the main login page

No, I am not. I now believe it is functioning as expected.

dcalhoun reviewed (commented):
When performing the testing instructions, I successfully authenticated in my browser, but when Studio launched, it did not connect my newly created WPCOM account. Is that expected when using the Calypso Live environment? 

The two analytic events did successfully log during my testing. 🎉

SiobhyB commented:
Thank you for testing this, @dcalhoun! 🙌 

> When performing the testing instructions, I successfully authenticated in my browser, but when Studio launched, it did not connect my newly created WPCOM account. Is that expected when using the Calypso Live environment?

The account should be authenticated using Calypso Live, I just tested and confirmed my newly created account was successfully connected after going through the flow:

https://github.com/Automattic/wp-calypso/assets/2998162/f650d69c-1f36-4fbd-b4fa-b45039146fe8

Could you share more about any varying factors? For example, were you already logged into Studio with a different account? I'd be grateful for any more specifics you can pinpoint so I can try to reproduce, I'll keep trying to reproduce on my own too. Thank you! 🙇‍♀️

dcalhoun reviewed (approved):
> Could you share more about any varying factors? For example, were you already logged into Studio with a different account?

Studio is, in fact, successfully updating to represent the newly created account. My first report was me unexpectedly encountering the absence of https://github.com/Automattic/studio/pull/86, which is not included in the 1.0.1 release I was testing. 😄 Sorry for the false report. 

I believe these changes are functioning as expected. 👏🏻

SiobhyB committed:
refactor: Update deprecated selector

`breakpoint-deprecated` was deprecated in favour of `break-mobile`. Source: https://github.com/Automattic/wp-calypso/pull/41985
